controllers/mirroring/mirroring_controller.go

```diff
@@ -0,0 +1,448 @@
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	// internalKey is a special key for client-mapping-config to establish mirroring between blockPools for internal mode
+	internalKey        = "internal"
+	mirroringFinalizer = "mirroring.ocs.openshift.io"
+)
+
+// MirroringReconciler reconciles a Mirroring fields for Ceph Object(s)
+// nolint:revive
+type MirroringReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+
+	log                 logr.Logger
+	ctx                 context.Context
+	clientMappingConfig *corev1.ConfigMap
+	storageClusterPeer  *ocsv1.StorageClusterPeer
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *MirroringReconciler) SetupWithManager(mgr ctrl.Manager) error {
+
+	if err := mgr.GetCache().IndexField(
+		context.TODO(),
+		&ocsv1alpha1.StorageConsumer{},
+		"annotations",
+		util.AnnotationIndexFieldFunc,
+	); err != nil {
+		return fmt.Errorf("unable to set up FieldIndexer on StorageConsumer for annotations: %v", err)
+	}
+
+	if err := mgr.GetCache().IndexField(
+		context.TODO(),
+		&ocsv1alpha1.StorageConsumer{},
+		"clientID",
+		util.StorageConsumerClientIDIndexFieldFunc,
+	); err != nil {
+		return fmt.Errorf("unable to set up FieldIndexer on StorageConsumer for clientID: %v", err)
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.ConfigMap{}, builder.WithPredicates(util.NamePredicate(storageclusterpeer.StorageClientMappingConfigName))).
+		Watches(&ocsv1.StorageClusterPeer{}, &handler.EnqueueRequestForObject{}).
+		Watches(&ocsv1alpha1.StorageConsumer{}, &handler.EnqueueRequestForObject{}).
+		Complete(r)
+}
+
+//+kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclusterpeers;storageconsumers,verbs=get;list;watch
+//+kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;watch
+//+kubebuilder:rbac:groups=core,resources=configmaps/finalizers,verbs=update
+//+kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;create;update;delete
+// +kubebuilder:rbac:groups=ceph.rook.io,resources=cephrbdmirrors,verbs=get;list;watch;create;update;delete
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+func (r *MirroringReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	var err error
+	r.ctx = ctx
+	r.log = log.FromContext(ctx, "Mirroring Controller", request)
+	r.log.Info("Reconciling Mirroring Controller.")
+
+	r.clientMappingConfig = &corev1.ConfigMap{}
+	r.clientMappingConfig.Name = request.Name
+	r.clientMappingConfig.Namespace = request.Namespace
+
+	if err = r.get(r.clientMappingConfig); err != nil {
+		if k8serrors.IsNotFound(err) {
+			r.log.Info("ClientMappingConfig %s not found. Ignoring since object must be deleted.")
+			return ctrl.Result{}, nil
+		}
+		r.log.Error(err, "Failed to get ConfigMap.", "ConfigMap", r.clientMappingConfig.Name)
```

Comment:
One thing that can help here is to add the Config Name to the log context so it will be part of every log message
Also, use the full NamespacedName form the request for that